service/src/main/resources/db/changesets/20221116_cohort_tables.yaml

```diff
@@ -0,0 +1,166 @@
+databaseChangeLog:
+  - changeSet:
+      id: initial_cohort_tables
+      author: marikomedlock
+      changes:
+        - createTable:
+            tableName: cohort
+            columns:
+              - column:
+                  name: study_id
+                  type: text
+                  constraints:
+                    references: study(study_id)
+                    foreignKeyName: fk_cohort_study_id
+                    nullable: false
+                    deleteCascade: true
+                  remarks: Deleting a study will cascade to delete the cohorts contained in it
+              - column:
+                  name: cohort_id
+                  type: text
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+                    unique: true
+              - column:
+                  name: underlay_name
+                  type: text
+                  constraints:
+                    nullable: false
+                    unique: false
+              - column:
+                  name: user_facing_cohort_id
+                  type: text
+                  constraints:
+                    nullable: false
+                    unique: false
+              - column:
+                  name: version
+                  type: integer
+                  constraints:
+                    nullable: false
+                    unique: false
+              - column:
+                  name: is_most_recent
```

Comment:
Rather than sometimes having the most recent cohort being in a non-editable state, how about a single flag `is_frozen`. Each cohort would have one active version with `is_frozen = false`. Creating a review would make a copy of the active review but with `is_frozen = true`.

Reply:
That's a valid approach, too and the one I originally took. I changed it because I think it's slightly cleaner to:
- Only create a new cohort row when required (i.e. on update cohort, instead of on create review)
- Check if something is the most recent version with an explicit flag rather than comparing timestamps (i.e. `is_most_recent=true`, instead of `last_modified=MAX(select last_modified from cohort where user_facing_cohort_id=11)`. I then thought about adding another flag `has_been_edited`, but that's back to 2 boolean flags like I've got here.